cdk/domino_cdk/provisioners/s3.py

```diff
@@ -25,55 +31,120 @@ def __init__(self, parent: cdk.Construct, construct_id: str, name: str, s3: List
         self.provision_buckets(name, s3)
         self.provision_iam_policy(name)
 
-    def provision_buckets(self, name: str, s3: List[config.S3]):
-        self.buckets = {}
-        for bucket, attrs in s3.buckets.items():
-            use_sse_kms_key = False
-            if attrs.sse_kms_key_id:
-                use_sse_kms_key = True
-                sse_kms_key = Key.from_key_arn(self, f"{bucket}-kms-key", attrs.sse_kms_key_id)
-
-            self.buckets[bucket] = Bucket(
-                self.scope,
-                bucket,
-                bucket_name=f"{name}-{bucket}",
-                auto_delete_objects=attrs.auto_delete_objects and attrs.removal_policy_destroy,
-                removal_policy=cdk.RemovalPolicy.DESTROY if attrs.removal_policy_destroy else cdk.RemovalPolicy.RETAIN,
-                enforce_ssl=True,
-                bucket_key_enabled=use_sse_kms_key,
-                encryption_key=(sse_kms_key if use_sse_kms_key else None),
-                encryption=(BucketEncryption.KMS if use_sse_kms_key else BucketEncryption.S3_MANAGED),
-            )
-            self.buckets[bucket].add_to_resource_policy(
+    def _provision_bucket(
+        self,
+        name: str,
+        bucket_id: str,
+        attrs: config.S3.Bucket,
+        server_access_logs_bucket: Optional[Bucket] = None,
+        **kwargs,
+    ) -> Bucket:
+        use_sse_kms_key = False
+        if attrs.sse_kms_key_id:
+            use_sse_kms_key = True
+            # TODO: is this correct???
+            sse_kms_key = Key.from_key_arn(self.scope, f"{bucket_id}-kms-key", attrs.sse_kms_key_id)
```

Comment:
this line used to be `Key.from_key_arn(self, f"{bucket}-kms-key", attrs.sse_kms_key_id)`, which had mypy complaining...